lms/djangoapps/teams/tests/test_views.py

```diff
@@ -523,3 +582,226 @@ def test_invalid_course_id(self, course_id):
+            user='student_enrolled_both_courses_other_team'
+        )
+        self.assertIn('already a member', json.loads(response.content)['developer_message'])
+
+    @ddt.data('staff', 'course_staff')
+    def test_not_enrolled_in_team_course(self, user):
+        response = self.post_create_membership(
+            400,
+            self.build_membership_data('student_unenrolled', self.test_team_1),
+            user=user
+        )
+        self.assertIn('not enrolled', json.loads(response.content)['developer_message'])
+
+
+@ddt.ddt
+class TestDetailMembershipAPI(TeamAPITestCase):
+    """Test cases for the membership detail endpoint."""
+
+    @ddt.data(
+        (None, 403),
+        ('student_inactive', 403),
+        ('student_unenrolled', 404),
+        ('student_enrolled_not_on_team', 200),
+        ('student_enrolled', 200),
+        ('staff', 200),
+        ('course_staff', 200),
+    )
+    @ddt.unpack
+    def test_access(self, user, status):
+        self.get_membership_detail(
+            self.test_team_1.team_id,
+            self.users['student_enrolled'].username,
+            status,
+            user=user
+        )
+
+    def test_bad_team(self):
+        self.get_membership_detail('foobar', self.users['student_enrolled'].username, 404)
+
+    def test_bad_username(self):
+        self.get_membership_detail(self.test_team_1.team_id, 'foobar', 404)
+
+    def test_no_membership(self):
+        self.get_membership_detail(
+            self.test_team_1.team_id,
+            self.users['student_enrolled_not_on_team'].username,
+            404
+        )
+
+    def test_expand_user(self):
+        result = self.get_membership_detail(
+            self.test_team_1.team_id,
+            self.users['student_enrolled'].username,
+            200,
+            {'expand': 'user'}
+        )
+        self.verify_expanded_user(result['user'])
+
+    def test_expand_team(self):
+        result = self.get_membership_detail(
+            self.test_team_1.team_id,
+            self.users['student_enrolled'].username,
+            200,
+            {'expand': 'team'}
+        )
+        self.verify_expanded_team(result['team'])
+
+
+@ddt.ddt
+class TestDeleteMembershipAPI(TeamAPITestCase):
+    """Test cases for the membership deletion endpoint."""
+
+    @ddt.data(
+        (None, 403),
+        ('student_inactive', 403),
+        ('student_unenrolled', 404),
+        ('student_enrolled_not_on_team', 404),
+        ('student_enrolled', 204),
+        ('staff', 204),
+        ('course_staff', 204),
+    )
+    @ddt.unpack
+    def test_access(self, user, status):
+        self.delete_membership(
+            self.test_team_1.team_id,
+            self.users['student_enrolled'].username,
+            status,
+            user=user
+        )
+
+    def test_bad_team(self):
+        self.delete_membership('foobar', self.users['student_enrolled'].username, 404)
```

Comment:
Very minor nit: I use a convention of ``no_such_team``, ``no_such_user`` etc. ``foobar`` doesn't jump out at me as being something that the test might not have set up for some reason.